src/fe-gtk/maingui.c

```diff
@@ -1084,6 +1084,12 @@ mg_tab_close_cb (GtkWidget *dialog, gint arg1, session *sess)
 	}
 }
 
+static struct {
+    server *server;
+    char channel[CHANLEN];
+    char key[64];
+} _last_chan = { 0 };
```

Comment:
Probably not in the maingui. I don't really know the ideal spot but for example the lastact feature in fkeys.c shoved a list in `hexchat.c`, don't know how awful that location is too.